vlayer/ethereum/circuits/lib/STORAGE_LAYOUT.md

```diff
@@ -0,0 +1,75 @@
+# Solidity storage layout
+
+EVM storage is a mapping from a number of `bytes32` type to another `bytes32` value. Each of 2<sup>256</sup> values is initialized to `0` by default. It's a kind of long-term memory used as smart contracts' storage. It allows contracts to use variables that aren't cleaned or removed after a function or transaction finish execution.
```

Comment:
It's does not need to be a number. In general it's bytes32 to bytes32. The key semantically can be a number, an address or a hash. Or anithing else that fits in bytes32 like string